code/modules/client/preferences/_preference.dm

```diff
@@ -312,7 +312,8 @@ GLOBAL_LIST_INIT(preference_entries_by_key, init_preference_entries_by_key())
 
 /// Returns whether or not this preference is accessible.
 /// If FALSE, will not show in the UI and will not be editable (by update_preference).
-/datum/preference/proc/is_accessible(datum/preferences/preferences)
+/// The applying_preference paramter designates whether the preference is being edited or applied
+/datum/preference/proc/is_accessible(datum/preferences/preferences, applying_preference=FALSE)
```

Comment:
I don't understand the point of this, can you explain this to me?

`is_accessible` should only be cared about in context to the UI.

Reply:
Preferences are being applied even if they aren't accessible. This prevents them from being applied if they aren't accessible.

For example, even if you didn't have the heterochromia quirk it would still apply the preference to the target with the previous value

Reply:
You shouldn't do this, you should do what the other prefs do and just have apply_to_human check if they should proceed. I don't think we need more flips and branches in prefs code, it's already quite chock full of them for lots of edge cases.

Reply:
The issue with that is target doesnt have a client because preferences are applied before the client is moved into the target

Reply:
I'd much rather preferences be passed through apply_to_human or similar then, I really don't like seeing branch parameters like this in 95% of places.

Reply:
Or actually, you should just do this like the preferences where:
1. You can change your style of nearsighted glasses (nearsighted quirk)
2. You can change your phobia (phobia quirk)

...so you don't need to muck with prefs code like this.

Reply:
> I'd much rather preferences be passed through apply_to_human or similar then, I really don't like seeing branch parameters like this in 95% of places.

Sure I can do that instead.

Reply:
Please try to do what nearsighted/phobia do if you can